src/xenia/kernel/xam/user_profile.cc

```diff
@@ -230,6 +236,199 @@ UserProfile::UserProfile(uint8_t index) {
+                                X_ONLINE_FRIENDSTATE_FLAG_PLAYING;
+  const uint32_t user_state = X_ONLINE_FRIENDSTATE_ENUM_ONLINE;
+
+  dummy_friend.xuid = GenerateOnlineXUID();
+  dummy_friend.session_id = XNKID();
+  dummy_friend.state = player_state | user_state;
+
+  // 5841125A want non-null session id to join session
+  xe::be<uint64_t> session_id = 0xAE00FFFFFFFFFFFF;
+  memcpy(dummy_friend.session_id.ab, &session_id, sizeof(XNKID));
+
+  // uint64_t xnkidInvite = 0xAE00FFFFFFFFFFFF;
+  // memcpy(dummy_friend.xnkidInvite.ab, &xnkidInvite, sizeof(XNKID));
+
+  std::string gamertag = fmt::format("Player {}", dist(gen));
+  std::u16string rich_presence = u"Playing on Xenia";
+
+  char* gamertag_ptr = reinterpret_cast<char*>(dummy_friend.Gamertag);
+  strcpy(gamertag_ptr, gamertag.c_str());
+
+  char16_t* rich_presence_ptr =
+      reinterpret_cast<char16_t*>(dummy_friend.wszRichPresence);
+  xe::string_util::copy_and_swap_truncating(
+      rich_presence_ptr, rich_presence, sizeof(dummy_friend.wszRichPresence));
+
+  dummy_friend.cchRichPresence =
+      static_cast<uint32_t>(rich_presence.size() * sizeof(char16_t));
+
+  return dummy_friend;
+}
+
+void UserProfile::AddDummyFriends(uint32_t friends_count) {
+  if (friends_.size() < X_ONLINE_MAX_FRIENDS) {
+    for (uint32_t i = 0; i < friends_count; i++) {
+      X_ONLINE_FRIEND peer = GenerateDummyFriend();
+
+      AddFriend(&peer);
+    }
+  }
+}
+
+bool UserProfile::GetFriendPresenceFromXUID(uint64_t xuid,
+                                            X_ONLINE_PRESENCE* presence) {
+  X_ONLINE_FRIEND peer = {};
+
+  const bool is_friend = GetFriendFromXUID(xuid, &peer);
+
+  if (!is_friend || presence == nullptr) {
+    return false;
+  }
+
+  presence->title_id = peer.title_id;
+  presence->state = peer.state;
+  presence->xuid = peer.xuid;
+  presence->session_id = peer.session_id;
+  presence->cchRichPresence = peer.cchRichPresence;
+
+  memcpy(presence->wszRichPresence, peer.wszRichPresence,
+         presence->cchRichPresence);
+
+  return true;
+}
+
+bool UserProfile::SetFriend(X_ONLINE_FRIEND update_peer) {
+  auto it = std::find_if(
+      friends_.begin(), friends_.end(), [&update_peer](X_ONLINE_FRIEND& peer) {
+        if (peer.xuid == update_peer.xuid) {
+          memcpy(&peer, &update_peer, sizeof(X_ONLINE_FRIEND));
+          return true;
+        }
+
+        return false;
+      });
+
+  if (it != friends_.end()) {
+    return false;
+  }
+
+  return true;
+}
+
+bool UserProfile::AddFriendFromXUID(uint64_t xuid) {
+  X_ONLINE_FRIEND peer = X_ONLINE_FRIEND();
+  peer.xuid = xuid;
+
+  return AddFriend(&peer);
+}
+
+bool UserProfile::AddFriend(X_ONLINE_FRIEND* peer) {
+  if (friends_.size() >= X_ONLINE_MAX_FRIENDS) {
+    return false;
+  }
+
+  if (peer == nullptr) {
+    return false;
+  }
+
+  if (IsFriend(peer->xuid)) {
+    return true;
+  }
+
+  XELOGI("{}", fmt::format("Added {:016X}", peer->xuid));
```

Comment:
No need to use fmt::format. You can just write XELOGI("Added {:016X}", peer->xuid)